evm/src/cpu/kernel/tests/mpt/load.rs

```diff
@@ -237,3 +242,41 @@ fn load_all_mpts_ext_to_leaf() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn load_mpt_txn_trie() -> Result<()> {
+    let load_all_mpts = KERNEL.global_labels["load_all_mpts"];
+
+    let txn = hex!("f860010a830186a094095e7baea6a6c7c4c2dfeb977efac326af552e89808025a04a223955b0bd3827e3740a9a427d0ea43beb5bafa44a0204bf0a3306c8219f7ba0502c32d78f233e9e7ce9f5df3b576556d5d49731e0678fd5a068cdf359557b5b").to_vec();
+
+    let trie_inputs = TrieInputs {
+        state_trie: Default::default(),
+        transactions_trie: HashedPartialTrie::from(Node::Leaf {
+            nibbles: Nibbles::from_str("0x80").unwrap(),
+            value: txn.clone(),
+        }),
+        receipts_trie: Default::default(),
+        storage_tries: vec![],
+    };
+
+    let initial_stack = vec![0xDEADBEEFu32.into()];
+    let mut interpreter = Interpreter::new_with_kernel(load_all_mpts, initial_stack);
+    interpreter.generation_state.mpt_prover_inputs = all_mpt_prover_inputs_reversed(&trie_inputs);
+    interpreter.run()?;
+    assert_eq!(interpreter.stack(), vec![]);
+
+    let mut expected_trie_data = vec![
+        0.into(),
+        U256::from(PartialTrieType::Leaf as u32),
+        2.into(),
+        128.into(), // Nibble
+        5.into(),   // value_ptr
+        txn.len().into(),
+    ];
+    expected_trie_data.extend::<Vec<U256>>(txn.into_iter().map(|b| b.into()).collect());
```

Comment:
nit: Can also do `expected_trie_data.extend(txn.into_iter().map(|b| U256:.from(b)))` to avoid the collect.